src/main/java/com/exactpro/th2/taskutils/BlockingScheduledRetryableTaskQueue.java

```diff
@@ -32,7 +33,7 @@ public class BlockingScheduledRetryableTaskQueue<V> {
     private final Queue<ScheduledRetryableTask<V>> taskQueue;
     private final Set<ScheduledRetryableTask<V>> taskSet;
 
-    private volatile long dataSize;
+    private final AtomicLong dataSize;
```

Comment:
question: Why do we need an AtomicLong here? We already access the `dataSize` field under the lock

Reply:
So, you right, in this case volatile isn't required in this case too for maxDataSize and dataSize